src/rgw/rgw_pubsub.cc

```diff
@@ -473,11 +474,26 @@ int RGWPubSub::get_topics(rgw_pubsub_topics *result)
 
 int RGWPubSub::Bucket::read_topics(rgw_pubsub_bucket_topics *result, RGWObjVersionTracker *objv_tracker)
 {
+  //read from cache
+  auto cache = ps->getRados()->get_topic_cache();
+  std::string key = bucket_meta_obj.pool.to_str() + bucket_meta_obj.oid;
+  if (auto e = cache->find(key)) {
+    *result = e->info;
+    return 0;
+  }
+  rgw_cache_entry_info cache_info;
```

Comment:
this needs to be passed to the `ps->read()` call below

Reply:
why do we need the `cache_info` definition here?
can we move inside the `if (result)` block?